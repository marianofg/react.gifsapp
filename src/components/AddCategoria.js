import React, { useState } from 'react'
import PropTypes from 'prop-types'

const AddCategoria = ({setCategoria}) => {

    const [inputvalue, setInputvalue] = useState('');

    const ingresarvalor = (e) => {
        setInputvalue(e.target.value)
    };

    const AgregarCategoria = (e) => {
        e.preventDefault();
        if (inputvalue.trim().length > 2) {
            setCategoria(cat => [inputvalue,...cat]);
            setInputvalue('');
        }        
    };
    return (
        <form onSubmit={AgregarCategoria}>
            <input
                type="text"
                value={inputvalue}
                onChange={ingresarvalor}
                placeholder="Buscar Gifs"
                ></input>
        </form>
    )
};

AddCategoria.propTypes = {
    setCategoria: PropTypes.func.isRequired
}

export default AddCategoria