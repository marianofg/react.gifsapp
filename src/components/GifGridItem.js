import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({id, titulo, url}) => {
  return (
    <div className='card animate__animated animate__bounce'>
        <img src={url} alt= {titulo}/>
            <p>{titulo}</p>
    </div>
  )
}

GifGridItem.propTypes = {
    titulo: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

export default GifGridItem