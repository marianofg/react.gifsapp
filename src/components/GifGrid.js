import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import GifGridItem from './GifGridItem';
import {getGifs} from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({ categoria }) => {

    
    const {data, loading} = useFetchGifs(categoria);


    return (
        <>
        <h3>{categoria}</h3>
        {loading && <p> cargando</p>}
        <div className='card-grid'>
                {data.map(({id,titulo,url}) =>{
                    return <GifGridItem key={id} titulo={titulo} url={url}></GifGridItem>
                })}
        </div>
        </>
    )
}

GifGrid.propTypes = {}

export default GifGrid