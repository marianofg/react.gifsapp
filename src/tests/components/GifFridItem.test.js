import {shallow} from 'enzyme'
import {GifGridItem} from "../../components/GifGridItem"

describe( 'Pruebas en GifGridItem',() =>{

    const titulo = 'un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem titulo={titulo} url={url}></GifGridItem>)
    test('mostramos el coponente', () => { 
        
        expect(wrapper).toMatchSnapshot();
     })

     test('titulo en un parrafo', () => { 
         const p = wrapper.find('p');
         expect(p.text().trim()).toBe(titulo);
      })
})
