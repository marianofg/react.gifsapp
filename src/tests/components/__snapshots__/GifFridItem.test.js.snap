// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem mostramos el coponente 1`] = `ShallowWrapper {}`;
